### ISSUE TITLE
Work Package Templates - Get All Templates Endpoint

### ISSUE THREAD

**Issue by RChandler234** [Northeastern-Electric-Racing/FinishLine#2156]
### Description

Blocked By #2148 

We need an endpoint so that users can get all the work package templates in the database. 

### Acceptance Criteria

- [ ] Add all these things in the work package files
- [ ] This endpoint should be on the route '/template'
- [ ] Anyone other than guests should be able to get all work package template
- [ ] Backend tests will be added in another ticket

### Proposed Solution

Look at other get all endpoints for an example
